plugins/sentinel/sentinel/mapping/sentinel_incident.py

```diff
@@ -0,0 +1,258 @@
+"""
+Sentiel Incident mapper for converting Microsoft Sentinel Sentiel Incidents to Howler bundles.
+"""
+import logging
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+from dateutil import parser
+
+# Use standard logging for now
+logger = logging.getLogger(__name__)
+
+
+class SentinelIncident:
+    """Class to handle mapping of Sentiel Incidents to Howler bundles."""
+
+    DEFAULT_CUSTOMER_NAME = "Unknown Customer"
+
+    def __init__(self, tid_mapping: Optional[Dict[str, str]] = None):
+        """
+        Initialize the Sentiel Incident mapper.
+
+        Args:
+            tid_mapping: Mapping of tenant IDs to customer names
+        """
+        self.tid_mapping = tid_mapping or {}
+
+    # --- Public mapping methods ---
+
+    def map_incident_to_bundle(self, sentinel_incident: Dict[str, Any]) -> Optional[Dict[str, Any]]:
+        """
+        Map an Sentiel Incident to a Howler bundle.
+
+        Args:
+            sentinel_incident (Dict[str, Any]): The Sentiel Incident data from Microsoft Graph.
+
+        Returns:
+            Optional[Dict[str, Any]]: Mapped bundle dictionary or None if mapping fails.
+
+        Example:
+            >>> mapper = SentinelIncident()
+            >>> bundle = mapper.map_incident_to_bundle(sentinel_incident)
+        """
+        try:
+            if not sentinel_incident:
+                logger.error("Empty incident data provided")
+                return None
+
+            required_fields = ["id", "tenantId", "createdDateTime"]
+            for field in required_fields:
+                if not sentinel_incident.get(field):
+                    logger.warning(f"Missing required field '{field}' in incident: {sentinel_incident}")
```

Comment:
Python loggers are optimized to use formatting strings, so it's "good practice" to use that approach (also for consistency across the codebase)

```suggestion
                    logger.warning(f"Missing required field '%s' in incident: %s", field, sentinel_incident)
```

https://docs.python.org/3/howto/logging.html#optimization